src/charm.py

```diff
@@ -102,6 +103,30 @@ def __init__(self, *args):
         super().__init__(*args)
 
         self.secrets = SecretCache(self)
+        self.peer_relation_app = DataPeer(
+            self,
+            relation_name=PEER,
+            additional_secret_fields=[
+                "monitoring-password",
+                "operator-password",
+                "replication-password",
+                "rewind-password",
+            ],
+            secret_field_name=SECRET_INTERNAL_LABEL,
+            deleted_label=SECRET_DELETED_LABEL,
+        )
+        self.peer_relation_unit = DataPeerUnit(
+            self,
+            relation_name=PEER,
+            additional_secret_fields=[
+                "monitoring-password",
+                "operator-password",
+                "replication-password",
+                "rewind-password",
+            ],
+            secret_field_name=SECRET_INTERNAL_LABEL,
+            deleted_label=SECRET_DELETED_LABEL,
+        )
```

Comment:
Is having those secrets (password) in both `peer_relation_app` and `peer_relation_unit` necessary?

Reply:
No, not at all :sweat_smile: 
I was just cutting corners in the POC code :sweat_smile: 

(I'm leaving the conversation up until I'd fix the issue.)